File: src/main/scala/com/strings/leetcode/string/Problem_1221_balancedStringSplit.scala
package com.strings.leetcode.string

object Problem_1221_balancedStringSplit {

  def main(args: Array[String]): Unit = {
    val s = "LRLRLLRRRL"
    println(balancedStringSplit(s))
  }

  def balancedStringSplit(s:String)={
    var num:Int = 0
    var res:Int = 0
    for(c <- s.toCharArray){
      if(c == 'L') num = num + 1
      else num = num - 1
      if(num == 0) res = res + 1
    }
    res
  }

}

File: src/main/java/leetcode/array/Problem_1221_balancedStringSplit.java
package leetcode.array;

public class Problem_1221_balancedStringSplit {

    public static void main(String[] args) {
        String s = "RLRRLLRLRL";
        System.out.println(balancedStringSplit(s));
    }

    public static int balancedStringSplit(String s){
        int num = 0;
        int res = 0;
        for(char c: s.toCharArray()){
            if(c == 'L') num ++;
            else num --;
            if(num == 0) res ++;
        }
        return res;
    }

}
